litestar/contrib/sqlalchemy/repository/_async.py

```diff
@@ -31,16 +31,31 @@ class SQLAlchemyAsyncRepository(AbstractAsyncRepository[ModelT], Generic[ModelT]
 
     match_fields: list[str] | str | None = None
 
-    def __init__(self, *, statement: Select[tuple[ModelT]] | None = None, session: AsyncSession, **kwargs: Any) -> None:
+    def __init__(
+        self,
+        *,
+        statement: Select[tuple[ModelT]] | None = None,
+        session: AsyncSession,
+        auto_expunge: bool = True,
```

Comment:
Repo level config for this stuff is the right move IMO.

My only suggestion would be to consider whether the expunge call should be default, or opt-in. Once litestar is out of beta this won't be something that can be easily reversed.

Sorry I can't do a more thorough review of the PR.

Reply:
In that case, perhaps we should disable `expunge` by default.  It can be a bit confusing when you hit edge cases like this.

Reply:
After a bit more thought, I think this is the right choice.  If you want the additional protection that it offers, it's there to use, otherwise, you can expect it to work in all cases like this.